src/jsd/el5042.cc

```diff
@@ -0,0 +1,198 @@
+      disable_status_bits_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      invert_checksum_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      checksum_polynomial_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      supply_voltage_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      clock_frequency_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      gray_code_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      multiturn_bits_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      singleturn_bits_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      offset_bits_node.size() != JSD_EL5042_NUM_CHANNELS ||
+      ssi_mode_node.size() != JSD_EL5042_NUM_CHANNELS) {
+    return false;
+  }
+
+  int ii;
+  for (ii = 0; ii < JSD_EL5042_NUM_CHANNELS; ii++) {
+    jsd_slave_config_.el5042.invert_feedback_direction[ii] =
+        invert_feedback_direction_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.disable_status_bits[ii] =
+        disable_status_bits_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.invert_checksum[ii] =
+        invert_checksum_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.checksum_polynomial[ii] =
+        checksum_polynomial_node[ii].as<uint32_t>();
+
+    jsd_slave_config_.el5042.supply_voltage[ii] =
+        supply_voltage_node[ii].as<uint8_t>();
+
+
+    jsd_slave_config_.el5042.gray_code[ii] =
+        gray_code_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.multiturn_bits[ii] =
+        multiturn_bits_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.singleturn_bits[ii] =
+        singleturn_bits_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.offset_bits[ii] =
+        offset_bits_node[ii].as<uint8_t>();
+
+    jsd_slave_config_.el5042.ssi_mode[ii] =
+        ssi_mode_node[ii].as<uint8_t>();
+
+    if (!ClockFrequencyFromString(clock_frequency_strings_[ii],
+                           jsd_slave_config_.el5042.clock_frequency[ii])) {
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool fastcat::El5042::Read()
+{
+  jsd_el5042_read((jsd_t*)context_, slave_id_);
+
+  const jsd_el5042_state_t* jsd_state =
+      jsd_el5042_get_state((jsd_t*)context_, slave_id_);
+
+  state_->el5042_state.position_ch1 = jsd_state->position[0];
+  state_->el5042_state.warning_ch1  = jsd_state->warning[0];
+  state_->el5042_state.error_ch1    = jsd_state->error[0];
+  state_->el5042_state.ready_ch1    = jsd_state->ready[0];
+  state_->el5042_state.position_ch2 = jsd_state->position[1];
+  state_->el5042_state.warning_ch2  = jsd_state->warning[1];
+  state_->el5042_state.error_ch2    = jsd_state->error[1];
+  state_->el5042_state.ready_ch2    = jsd_state->ready[1];
+
+  return true;
+}
+
+bool fastcat::El5042::ClockFrequencyFromString(std::string           clock_frequency_string,
+                                               jsd_el5042_clock_t&   clock_frequency)
+{
+  MSG_DEBUG("Parsing clock frequency string: %s", clock_frequency_string.c_str());
+
+  if (clock_frequency_string.compare("10MHz") == 0) {
```

Comment:
Have you been able to test the different read rates using this interface on hardware?